AnalyteExplorer/src/client/AnalyteExplorer/components/DownloadPage.tsx

```diff
@@ -0,0 +1,338 @@
+
+        return result;
+      } catch (err) {
+        console.log(err);
+        return undefined;
+      }
+    };
+
+    const processChartMetaDataError = (analyteName: string, error: string) => {
+      processChartMetaData(analyteName, error);
+    };
+
+    const callGeneAPI = async (gene: string) => {
+      const apiGeneID = await getGeneID(gene);
+
+      if (apiGeneID !== "") {
+        const geneMetaData = await getGeneMetaData(apiGeneID);
+
+        if (geneMetaData !== undefined) {
+          const metaBoxTitle = `${analyteName}: ${capitalizeKebabCase(
+            geneMetaData["name"]
+          )}`;
+          const metaBoxBody = `${geneMetaData["summary"]}`;
+          const metaBoxSubTitle = `${capitalizeKebabCase(
+            geneMetaData["type_of_gene"]
+          )} // Aliases: ${geneMetaData["alias"]}`;
+          processChartMetaData(metaBoxTitle, metaBoxBody, metaBoxSubTitle);
+        } else {
+          processChartMetaDataError(analyteName, "UNABLE TO RETRIEVE METADATA");
+        }
+      } else {
+        processChartMetaDataError(analyteName, "NO METADATA FOUND");
+      }
+    };
+
+    const getMetaData = (type: string) => {
+      if (type === "blood transcription module") {
+        Query.executeSql({
+          containerPath: "/AnalyteExplorer",
+          schemaName: "lists",
+          sql: `SELECT DISTINCT id, name, genes
+                    FROM blood_transcription_modules
+                    WHERE blood_transcription_modules.id = '${analyteName}'
+                    `,
+          success: processBTMMetaData,
+          failure: processFailure,
+        });
+      } else if (analyteType === "gene") {
+        callGeneAPI(analyteName);
+      }
+    };
+
+    // grabs data for a specific analyte under specific set of disease conditions
+    const getData = () => {
+      Query.selectRows({
+        schemaName: "lists",
+        queryName: "gene_expression",
+        filterArray: [
+          Filter.create("analyte_id", analyteName.toUpperCase()),
+          Filter.create("condition", filters, Filter.Types.CONTAINS_ONE_OF),
+        ],
+        success: processData,
+        failure: processFailure,
+      });
+    };
+
+    if (
+      !isCancelled &&
+      analyteName !== "" &&
+      analyteType !== "" &&
+      filters.length > 0
+    ) {
+      console.log("fetching data...");
+
+      // wipe cached data before querying new data
+      rawData !== null ? setRawData(null) : null;
+      chartData !== null ? setChartData(null) : null;
+      chartMetadata !== null ? setChartMetadata(null) : null;
+      errorMsg !== "" ? setErrMsg("") : null;
+
+      getData();
+      getMetaData(analyteType);
+      console.log("meep");
+    }
+    return () => {
+      isCancelled = true;
+    };
+  }, [analyteName, analyteType, filters]);
+
+  const isDataLoaded =
+    rawData !== null && chartData !== null && chartMetadata !== null;
```

Comment:
Suggest React.useMemo here:
```suggestion
  const isDataLoaded = React.useMemo(() => 
    rawData !== null && chartData !== null && chartMetadata !== null
  , [rawData, chartData, chartMetadata]);
```